compiler/vm.nim

```diff
@@ -711,7 +711,7 @@ proc rawExecute(c: PCtx, start: int, tos: PStackFrame): TFullReg =
     of opcLdObj:
       # a = b.c
       decodeBC(rkNode)
-      let src = regs[rb].node
+      let src = if regs[rb].kind == rkNode: regs[rb].node else: regs[rb].nodeAddr[]
```

Comment:
Maybe something like this is better
```nim
      let kind = regs[rb].kind
      let src = 
        if kind == rkNode: regs[rb].node
        elif kind == rkNodeAddr: regs[rb].nodeAddr[]
        else: doAssert false, $kind
```

Reply:
The error shows up and can be localized without this and if the compile is release or danger it won't work, so it really doesn't add anything. So no, that's not better.